digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/client/RoadLinkChangeClient.scala

```diff
@@ -0,0 +1,168 @@
+    {
+      case JInt(bigIntValue) =>
+        AdministrativeClass(bigIntValue.toInt)
+      case JNull => Unknown
+    },
+    {
+      case adminClass: AdministrativeClass =>
+        JObject(JField("adminClass", JInt(adminClass.value)))
+    }
+  ))
+
+  object TrafficDirectionSerializer extends CustomSerializer[TrafficDirection](_ => (
+  {
+    case JInt(directionValue) =>
+      directionValue.toInt match {
+        case 0 => TrafficDirection.BothDirections
+        case 1 => TrafficDirection.TowardsDigitizing
+        case 2 => TrafficDirection.AgainstDigitizing
+        case _ => TrafficDirection.UnknownDirection
+      }
+  },
+  {
+    case trafficDirection: TrafficDirection =>
+      trafficDirection match {
+        case TrafficDirection.BothDirections => JInt(0)
+        case TrafficDirection.TowardsDigitizing => JInt(1)
+        case TrafficDirection.AgainstDigitizing => JInt(2)
+        case _ => JNull
+      }
+  }
+  ))
+
+  object GeometrySerializer extends CustomSerializer[List[Point]](_ => (
+    {
+      case JString(lineString) =>
+        lineStringToPoints(lineString)
+    },
+    {
+      case points: List[Point] =>
+        JObject(JField("geometry", JString(""))) // not implemented until reverse operation is needed
+    }
+  ))
+
+  implicit val formats = DefaultFormats + changeItemSerializer + RoadLinkChangeTypeSerializer + GeometrySerializer +
+    AdminClassSerializer + TrafficDirectionSerializer
+
+  case class RoadLinkInfo(linkId: String, linkLength: Double, geometry: List[Point], roadClass: Int, adminClass: AdministrativeClass, municipality: Int, trafficDirection: TrafficDirection)
+  case class ReplaceInfo(oldLinkId: String, newLinkId: String, oldFromMValue: Double, oldToMValue: Double, newFromMValue: Double, newToMValue: Double)
+  case class RoadLinkChange(changeType: RoadLinkChangeType, oldLink: Option[RoadLinkInfo], newLinks: Seq[RoadLinkInfo], replaceInfo: Seq[ReplaceInfo])
+
+  def fetchLatestSuccessfulUpdateDate(): DateTime = {
+    // placeholder value as long as fetching this date from db is possible
+    DateTime.parse("2022-05-10")
+  }
+
+  def listFilesAccordingToDates(since: DateTime, until: DateTime) = {
+    def isValidKey(key: String): Boolean = {
+      try {
+        val keyParts = key.replace(".json", "").split("_")
+        val keySince = DateTime.parse(keyParts.head)
+        val keyUntil = DateTime.parse(keyParts.last)
+        !(keySince.isBefore(since) || keyUntil.isAfter(until)) // get no changes before or after the requested period
+      } catch {
+        case illegalArgument: IllegalArgumentException =>
+          logger.error("Key provides no valid dates.")
+          false
+        case e: Exception =>
+          logger.error(e.getMessage)
+          false
+      }
+    }
+
+    val request = ListObjectsV2Request.builder().bucket(s3Bucket).build()
+    val objects = s3Client.listObjectsV2(request).contents().asScala.toList
```

Comment:
Olisko mahdollista tehdä AWSService metodi joka rakentaa objects, jotenkin näin: 
def (client: s3,bucketName: string) {
client.listObjectsV2(ListObjectsV2Request.builder().bucket(bucketName).build()).contents()
} 
Vältetään AWS sdk leviämistä ympäri koodi pohjaa.

Reply:
Hyvä pointti, eiköhän se onnistu.